### PR TITLE
[feat]: customer authentication

### PR DESCRIPTION
**Pull request by mog-rn**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,22 +2,40 @@
   "name": "songa.api",
   "version": "0.0.1",
   "description": "This is the backend application that will help run the backend of the Songa application",
-  "main": "./src/app.js",
+  "main": "./src/index.ts",
   "scripts": {
     "test": "test",
-    "start:dev": "nodemon ./src/app.js"
+    "start:dev": "nodemon ./src/index.ts",
+    "dev": "nodemon ./src/index.ts"
```
[the remark this answers is not in the document]
Comment (Martindevelops-songa) [reply package.json:9]: fixed

File: src/controllers/UserControllers.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import PasswordHash from "../helpers/PasswordHash";
+import { PrismaClient } from "@prisma/client";
+const prisma = new PrismaClient();
+
+//check user
+interface CheckUserProps {
+  phone: string;
+  email: string | undefined;
+}
+const checkUser = async ({ phone, email }: CheckUserProps) => {
```
[the remark this answers is not in the document]
Comment (Martindevelops-songa) [reply src/controllers/UserControllers.ts]: the email is there but not as a primary. some people would like to use email at times. just expanding the options.

**Review by Martindevelops-songa — COMMENTED** [review]
update

**Review by Martindevelops-songa — COMMENTED** [review]
updated

File: prisma/schema.prisma [hunk 3]
```diff
@@ -10,3 +10,21 @@ datasource db {
...
+  first_name String
+  last_name String
+  phone String @unique
+  password String
+  salt String
+  avatar String?
+  email String? @unique
+  address String?
+  gender Gender? 
+  id String @id @default(uuid())
```
Comment (dennisngeno-songa) [inline prisma/schema.prisma:29]: As part of your suggestion on how we can model the schema on whatsapp, I dont see this method 
```
@@map("customer care agents")```

File: prisma/schema.prisma [hunk 4]
```diff
@@ -10,3 +10,21 @@ datasource db {
...
+enum Gender{
+  M 
+  F 
+  Other
+}
+
+model User{
+  first_name String
+  last_name String
+  phone String @unique
```
[the remark this answers is not in the document]
Comment (dennisngeno-songa) [reply prisma/schema.prisma:22]: Should we do the same for Customer Care Support?

**Review by mog-rn — COMMENTED** [review]
LGTM but we need to make a few changes before merge.

File: prisma/schema.prisma (hunk 3 above)
Comment (Martindevelops-songa) [reply prisma/schema.prisma:29]: will fix it.

**Review by Martindevelops-songa — COMMENTED** [review]
comment

**Review by Martindevelops-songa — APPROVED** [review]
update package json

**Pull request merged**